src/player/Player.php

```diff
@@ -2092,22 +2092,25 @@ public function toggleSprint(bool $sprint) : bool{
 		return true;
 	}
 
-	public function toggleSneak(bool $sneak, bool $sneakPressed = true) : bool{
-		if($sneak === $this->sneaking && $sneakPressed === $this->sneakPressed){
+	public function toggleSneak(?bool $sneak, bool $sneakPressed = true) : bool{
```

Comment:
I don't think we should be changing this signature, especially not on the stable branch. You could do `?? $this->player->isSneaking()` in the packet handler instead, that should be sufficient and shouldn't require any changes to `Player`